nucliadb_texts2/src/query_io.rs

```diff
@@ -19,46 +19,42 @@
 
 use crate::schema::TextSchema;
 use nucliadb_core::query_language::{BooleanExpression, BooleanOperation, Operator};
-use tantivy::query::{AllQuery, BooleanQuery, Occur, Query, TermQuery};
+use tantivy::query::{AllQuery, BooleanQuery, Occur, PhraseQuery, Query, TermQuery};
 use tantivy::schema::{Facet, IndexRecordOption};
 use tantivy::Term;
 
-fn translate_literal_to_facet_query(literal: &str, schema: &TextSchema) -> Box<dyn Query> {
+fn translate_label_to_facet_query(literal: &str, schema: &TextSchema) -> Box<dyn Query> {
     let facet = Facet::from_text(literal).unwrap();
     let term = Term::from_facet(schema.facets, &facet);
     Box::new(TermQuery::new(term, IndexRecordOption::Basic))
 }
 
-fn translate_literal_to_text_query(literal: &str, schema: &TextSchema) -> Box<dyn Query> {
-    // If the literal has spaces, convert them to multiple queries with an AND operator
-    if literal.contains(' ') {
-        let mut operands: Vec<(Occur, Box<dyn Query>)> = Vec::new();
-        for word in literal.split_whitespace() {
-            let term = Term::from_field_text(schema.text, &word.to_lowercase());
-            operands.push((Occur::Must, Box::new(TermQuery::new(term, IndexRecordOption::Basic))));
-        }
-        return Box::new(BooleanQuery::new(operands));
+fn translate_keyword_to_text_query(literal: &str, schema: &TextSchema) -> Box<dyn Query> {
```

Comment:
This inverted conditions is maybe a bit confusing? I'd also put the alternative branch on an else, since they are the same level of importance. And we can reuse the term construction. So something like this may be nicer. What do you think?

```rust
    let terms = literal.split_whitespace().map(|w| Term::from_field_text(schema.text, &w.to_lowercase())).collect_vec();
    if literal.len() == 1 {
        Box::new(TermQuery::new(terms[0], IndexRecordOption::Basic))
    } else {
        Box::new(PhraseQuery::new(terms.collect()))
    }
```

Reply:
I like this code more indeed! but does not complie 😢 I'll fight a bit more with the compiler and see if I succeed.
Otherwise I'll leave it as is ok?

Reply:
made it work! 🚀